### PR TITLE
Added search component 

### LINKED ISSUE #11: Add search component
It would be great to have a search component that can be displayed in the /tags page or the /blog list.

### PR DESCRIPTION
**Pull request by Elijah-trillionz**
Fixes #11 issue

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module.exports = {
   content: ["./app/**/*.{js,jsx,ts,tsx}"],
   darkMode: "class",
+  important: true,
```
Comment (josemiguel-alvarez) [inline tailwind.config.js:4]: Why is this needed?

File: app/routes/blog/index.tsx [hunk 2]
```diff
@@ -4,6 +4,9 @@ import type { Frontmatter } from "~/utils/posts.server";
 import { getPostsSortedByDate } from "~/utils/posts.server";
 import { PostsList } from "~/components/PostsList";
 import { getPagingData } from "~/utils/paging.server";
+import { SearchForm } from "~/components/SearchForm";
+import { useEffect, useState } from "react";
```
Comment (josemiguel-alvarez) [inline app/routes/blog/index.tsx]: `useEffect` is not used and can be removed.

File: app/routes/blog/index.tsx [hunk 3]
```diff
@@ -23,12 +26,15 @@ export const loader = async ({ request }: { request: Request }) => {
 export default function Blog() {
   const { page, posts, nextPage, previousPage, totalPages } =
     useLoaderData<LoaderData>();
+  const [query, setQuery] = useState("");
+  const controlledPosts = useSearchQuery(posts, query);
```
Comment (josemiguel-alvarez) [inline app/routes/blog/index.tsx]: This is only filtering the posts that are already displayed in the screen. We should filter using all the posts available. 

Try duplicating the posts in the app to reproduce this.

**Comment by josemiguel-alvarez** [comment]
I like the design you used in the `/tags` page. What do you think about using the same design in the `/blog` page for consistency? The code could also be simplified.

File: app/routes/blog/index.tsx (hunk 3 above)
Comment (josemiguel-alvarez) [reply app/routes/blog/index.tsx]: This is a tutorial that shows a good way of doing it: https://dev.to/zachtylr21/building-a-simple-search-ui-with-remix-57da

Comment (Elijah-trillionz) [reply app/routes/blog/index.tsx]: Oh, just figured you must have filtered the posts already so as to display them in chunks.
Sorry, I didn't think of that earlier.

File: tailwind.config.js (hunk 1 above)
Comment (Elijah-trillionz) [reply tailwind.config.js:4]: for overriding the styles set by MUI, I have tried not using it

**Comment by Elijah-trillionz** [comment]
> I like the design you used in the `/tags` page. What do you think about using the same design in the `/blog` page for consistency? The code could also be simplified.

Yes, it is possible, but then I'd have to push the `h1` header to the left just like in the `/tags` page. Is this okay with you?

File: app/routes/blog/index.tsx (hunk 3 above)
Comment (josemiguel-alvarez) [reply app/routes/blog/index.tsx]: No problem Elijah!

**Comment by josemiguel-alvarez** [comment]
> > I like the design you used in the `/tags` page. What do you think about using the same design in the `/blog` page for consistency? The code could also be simplified.
> 
> Yes, it is possible, but then I'd have to push the `h1` header to the left just like in the `/tags` page. Is this okay with you?

Yes, perfect!

**Comment by josemiguel-alvarez** [comment]
@all-contributors please add @Elijah-trillionz  for code

**Comment by josemiguel-alvarez** [comment]
@all-contributors please add @Elijah-trillionz for design

**Comment by Elijah-trillionz** [comment]
There is a current bug in production mode on the search input.
I'm working on this now.

**Comment by Elijah-trillionz** [comment]
> There is a current bug in production mode on the search input. I'm working on this now.

bug fixed in new commit

**Review by josemiguel-alvarez — APPROVED** [review]
Nice work, thank you!
